mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt

```diff
@@ -28,4 +31,13 @@ class SocialFacade(private val auth: Auth, private val store: Store) {
       it.mapNotNull { doc -> doc?.toProfile() }
     }
   }
+  /**
+   * Follows the given user
+   *
+   * @param followed the user to follow.
+   */
+  suspend fun follow(followed: Person) {
+    val currentUid = auth.currentUser.firstOrNull()?.uid ?: ""
+    store.collection("users").document(followed.uid).update { arrayUnion("followers", currentUid) }
+  }
```

Comment:
This method requires access to the current user, so it should be added to `AuthenticatedUser`. Only an authenticated user may follow another user.

This also solves the problem of knowing what to do on errors, if `auth.currentUser.firstOrNull()` returns `null` for instance.